pipeline-templates/job/deploy.yml

```diff
@@ -0,0 +1,80 @@
+  pool:
+    name: DAS - Continuous Deployment Agents
+  environment: ${{ parameters.Environment }}
+  strategy:
+    runOnce:
+      deploy:
+        steps:
+        - template: azure-pipelines-templates/deploy/step/wait-azure-devops-deployment.yml@das-platform-building-blocks
+          parameters:
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            EnvironmentId: $(Environment.Id)
+            PipelineName: $(Build.DefinitionName)
+            RunId: $(Build.BuildId)
+        - template: azure-pipelines-templates/deploy/step/set-backendaccessrestrictions-variable.yml@das-platform-building-blocks
+          parameters:
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            SharedEnvResourceGroup: $(SharedEnvResourceGroup)
+            SharedEnvVirtualNetworkName : $(SharedEnvVirtualNetworkName)
+            BackEndAccessRestrictionsExcludedSubnets: $(BackEndAccessRestrictionsExcludedSubnets)
+            ResourceEnvironmentName: $(ResourceEnvironmentName)
+            UnrestrictedEnvironments: $(UnrestrictedEnvironments)
+            UptimeMonitoringAccessRestrictions: $(UptimeMonitoringAccessRestrictions)
+        - template: azure-pipelines-templates/deploy/step/arm-deploy.yml@das-platform-building-blocks
+          parameters:
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            SubscriptionId: ${{ parameters.SubscriptionId }}
+            Location: ${{ parameters.ResourceGroupLocation }}
+            Environment: ${{ parameters.Environment }}
+            TemplatePath: $(Pipeline.Workspace)/${{ parameters.SolutionBaseName }}/azure/template.json
+            ParametersPath: $(Pipeline.Workspace)/${{ parameters.SolutionBaseName }}/azure/template.parameters.json
+            IsMultiRepoCheckout: true
+            TemplateSecrets:
+              LoggingRedisConnectionString: ${{ parameters.LoggingRedisConnectionString }}
+              ConfigurationStorageConnectionString: ${{ parameters.ConfigurationStorageConnectionString }}
+              SharedStorageAccountConnectionString: ${{ parameters.SharedStorageAccountConnectionString }}
+        - template: azure-pipelines-templates/deploy/step/generate-config.yml@das-platform-building-blocks
+          parameters:
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            SourcePath: $(Pipeline.Workspace)/das-employer-config/Configuration/das-tasks
+            TargetFileName: '*.schema.json'
+            TableName: Configuration
+        - template: azure-pipelines-templates/deploy/step/app-deploy.yml@das-platform-building-blocks
+          parameters:
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            AppServiceName: $(ApiAppServiceName)
+            DeploymentPackagePath: $(Pipeline.Workspace)/${{ parameters.SolutionBaseName }}/${{ parameters.SolutionBaseName }}.API/${{ parameters.SolutionBaseName }}.API.zip
+        - task: AzureRmWebAppDeployment@4
```

Comment:
Is ZipDeploy deployment type not working for this app?

Reply:
Didn't work for the function app and is now broken for the web app as well, will try to identify why but might need to use and earlier version of the app-deploy template

Reply:
It doesn't build locally using dotnet core and looking back at the history in this [branch](https://github.com/SkillsFundingAgency/das-tasks/tree/DASD-7575-MigrateToAppServiceWebJob) (before I rebased and tidied up) it looks like I had a lot of problems getting that to work on the agent as well.  Publishing with `VSBuild@1` was the solution but this will mean using `webDeploy` instead of `zipDeploy`

Reply:
There is a solution for this by changing the arguments of the VsBuild Task. An example commit is here
 eg: https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/commit/90f6e6ad7a756c09a5fb43ff87cc4a6bb233bae6

Reply:
Build uses zipDeploy